schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/SchemaRegistryResource.java

```diff
@@ -1142,10 +1353,16 @@ public Response mergeSchemaVersion(@ApiParam(value = "Details about schema versi
     @Path("/schemas/branch/{branchId}")
     @ApiOperation(value = "Delete a branch give its name", tags = OPERATION_GROUP_SCHEMA)
     @UnitOfWork
-    public Response deleteSchemaBranch(@ApiParam(value = "Schema Branch Name", required = true) @PathParam("branchId") Long schemaBranchId) {
+    public Response deleteSchemaBranch(@ApiParam(value = "Schema Branch Name", required = true) @PathParam("branchId") Long schemaBranchId,
+                                       @Context SecurityContext securityContext) {
         try {
+            authorizationAgent.authorizeDeleteSchemaBranch(AuthorizationUtils.getUserAndGroups(securityContext),
+                    schemaRegistry, schemaBranchId);
             schemaRegistry.deleteSchemaBranch(schemaBranchId);
             return WSUtils.respond(Response.Status.OK);
+        } catch (AuthorizationException e) {
+            LOG.warn("Access denied. ", e);
```

Comment:
Let's encapsulate the body of **catch(AuthorizationException e)** in a function

Reply:
We can do this, but then we should do the same for all the other catch blocks.